src/server/pkg/storage/chunk/writer.go

```diff
@@ -26,105 +27,109 @@ const (
 // of the rolling hash function.
 var initialWindow = make([]byte, WindowSize)
 
-// WriterFunc is a callback that returns a data reference to the next chunk and the annotations within the chunk.
-type WriterFunc func(*DataRef, []*Annotation) error
+// WriterFunc is a callback that returns the annotations within a chunk.
+type WriterFunc func([]*Annotation) error
 
-// byteSet is a unit of work for the workers.
-// A worker will roll the rolling hash function across the bytes
+// dataSet is a unit of work for the workers.
+// A worker will roll the rolling hash function across the data set
 // while processing the associated annotations.
-type byteSet struct {
-	data        []byte
+type dataSet struct {
```

Comment:
This means that we now have `dataSet` and `datumSet`, couple that with the fact that dataset is already a well known word with a lot of meaning attached to it and I think this is not a good name for this struct.

Reply:
I mean, it is a data set. But, I do get that it does not tell you a lot. I am open to alternative names, I just don't know what a good one would be right now. I think naming in general has been a bit of a challenge with the storage layer (and other parts of Pachyderm).

Reply:
I guess it doesn't matter that much since it's an internal name. I think the first sentence of the comment might give some ideas though, it seems like the main purpose of this struct is to be a unit of work for a worker, so maybe calling it something like `workUnit` would work?